sklearn/cross_validation.py

```diff
@@ -595,15 +595,50 @@ def __len__(self):
 class BaseShuffleSplit(with_metaclass(ABCMeta)):
     """Base class for ShuffleSplit and StratifiedShuffleSplit"""
 
-    def __init__(self, n, n_iter=10, test_size=0.1, train_size=None,
-                 random_state=None):
-        self.n = n
-        self.n_iter = n_iter
-        self.test_size = test_size
-        self.train_size = train_size
-        self.random_state = random_state
-        self.n_train, self.n_test = _validate_shuffle_split(n, test_size,
-                                                            train_size)
+    def __init__(self, *args, **kwargs):
```

Comment:
don't do that. It makes following the code much harder. Why did you refactor like this?